clippy_lints/src/doc.rs

```diff
@@ -516,34 +518,54 @@ fn check_doc<'a, Events: Iterator<Item = (pulldown_cmark::Event<'a>, Range<usize
             Html(_html) => (),             // HTML is weird, just ignore it
             SoftBreak | HardBreak | TaskListMarker(_) | Code(_) | Rule => (),
             FootnoteReference(text) | Text(text) => {
-                if Some(&text) == in_link.as_ref() {
+                if Some(&text) == in_link.as_ref() || ticks_unbalanced {
                     // Probably a link of the form `<http://example.com>`
                     // Which are represented as a link to "http://example.com" with
                     // text "http://example.com" by pulldown-cmark
                     continue;
                 }
-                headers.safety |= in_heading && text.trim() == "Safety";
-                headers.errors |= in_heading && text.trim() == "Errors";
-                headers.panics |= in_heading && text.trim() == "Panics";
-                let index = match spans.binary_search_by(|c| c.0.cmp(&range.start)) {
-                    Ok(o) => o,
-                    Err(e) => e - 1,
-                };
-                let (begin, span) = spans[index];
-                if in_code {
-                    if is_rust {
-                        let edition = edition.unwrap_or_else(|| cx.tcx.sess.edition());
-                        check_code(cx, &text, edition, span);
-                    }
+                if text.contains('`') && !in_code {
+                    let (_, start) = spans[0];
+                    let span = if let Some((_, end)) = spans.last() {
+                        start.with_hi(end.hi())
+                    } else {
+                        start
+                    };
+                    span_lint(
+                        cx,
+                        DOC_MARKDOWN,
+                        span,
+                        "backticks are unbalanced; one may be missing a pair",
+                    );
+                    ticks_unbalanced = true;
                 } else {
-                    // Adjust for the beginning of the current `Event`
-                    let span = span.with_lo(span.lo() + BytePos::from_usize(range.start - begin));
-
-                    check_text(cx, valid_idents, &text, span);
+                    headers.safety |= in_heading && text.trim() == "Safety";
```

Comment:
The following is a copy from the previous implementation. I think it would be better to add a `continue` in line 540 and unindent the code again :upside_down_face: